plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -272,16 +272,21 @@ codeActionProvider ideState plId (CodeActionParams _ _ docId _ context) = Right
   where
 
     getCodeActions = do
-        diags <- getDiagnostics ideState
+        allDiags <- getDiagnostics ideState
         let docNfp = toNormalizedFilePath' <$> uriToFilePath' (docId ^. LSP.uri)
             numHintsInDoc = length
-              [d | (nfp, _, d) <- diags
+              [d | (nfp, _, d) <- allDiags
                  , validCommand d
                  , Just nfp == docNfp
               ]
+            numHintsInContext = length
+              [d | d <- diags
```

Comment:
Is this also assuming that the diagnostics we in the context are exclusively hlint hints? I'm pretty sure that's not true: you could get other warnings or errors as well.

Reply:
I got the impression that `validCommand` checks that, is it not the case?